### PR TITLE
Feature: Renting manager role

### PR DESCRIPTION
**Pull request by Ellen-Wittingen**
Added a role for renting managers with which they have the following permissions:
- Activities
  - See a list of all activities, also the ones that are locked
  - Create, edit, and delete activities
  - View an activity, including the tabs "Samenvatting", "Bestellingen", and "Producttotalen"
  - Access the order screen and place orders with it, but credit mutations (Geld inleggen)
- Invoices
  - See a list of all invoices
  - View an invoice, but not edit or send it
- Pricelists
  - See a list of all unarchived pricelists and the prices for each product, but not edit
- Users
  - See a list of all manually added users, but not from AMBER except for themselves
  - View a manually added user and their credit, orders, and credit mutations

The following permissions were changed for main bartenders (after talking to Soccie):
- Activities: In the activity view, only the tabs "Samenvatting" and "Producttotalen" can be viewed now.

I also added tests for the new renting-manager role and added some tests that were missing for some controllers. The syntax used is a bit inconsistent across tests (also in the already existing tests), I will fix this in some another pull request in the future.

To easily log in as the different roles, I added new users to the seeding of AMBER-API which are linked to SOFIA roles in the seeding of SOFIA, see [#412](https://github.com/csvalpha/amber-api/pull/412)

### CODE REVIEW

File: app/views/invoices/show.html.erb [hunk 1]
```diff
@@ -63,7 +63,7 @@
         </tbody>
       </table>
 
-      <% unless @invoice.paid? %>
+      <% unless !policy(Invoice).send_invoice? || @invoice.paid? %>
```
Comment (lodewiges) [inline app/views/invoices/show.html.erb:66]: What does this mean?
and why is the change needed?

Comment (Ellen-Wittingen) [reply app/views/invoices/show.html.erb:66]: A user with the renting manager role can also access this page. However, they are not allowed to send an invoice, so for them the url should not appear on the page. So we check here with the policy for invoices (described in app/policies/invoice_policy.rb) is the person accessing the page is allowed to send an invoice

Comment (Ellen-Wittingen) [reply app/views/invoices/show.html.erb:66]: Before this there was no renting manager role, so the check was unnecessary as only treasurers (who are allowed to send invoice) could access this page

File: app/javascript/components/activity/ProductTotals.vue [hunk 2]
```diff
@@ -71,6 +79,7 @@ export default {
       let params = {user: this.user.id, paid_with_cash: this.user.paid_with_cash, paid_with_pin: this.user.paid_with_pin};
       this.$http.get('/activities/'+this.activity+'/product_totals', { params }).then((response) => {
         this.orderTotals = response.body;
+        this.totalAmount = this.orderTotals.reduce((a, b) => a + parseFloat(b.price), 0.0);
```
Comment (lodewiges) [inline app/javascript/components/activity/ProductTotals.vue:82]: i am getting a warning trying to run this 
image

File: spec/controllers/price_lists_controller/create_spec.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+describe PriceListsController, type: :controller do
+  describe 'POST create' do
+    let(:price_list) do
+      build(:price_list)
+    end
+    let(:request) do
+      post :create, params: { price_list: price_list.attributes }
+    end
+
+    it 'treasurer cna create a new price_list' do
```
Comment (lodewiges) [inline spec/controllers/price_lists_controller/create_spec.rb:12]:
```suggestion
    it 'treasurer can create a new price_list' do
```

File: app/models/role.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 class Role < ApplicationRecord
-  enum role_type: { treasurer: 0, main_bartender: 1 }
+  enum role_type: { treasurer: 0, main_bartender: 1, renting_manager: 2 }
```
Comment (lodewiges) [inline app/models/role.rb:2]: do these numbers mean something

**Review by lodewiges — CHANGES_REQUESTED** [review]
I tested and it looks good, you said in the describtion test are not writen in the same way could you elobrate on that so it can be fixed in the future. Sorry it took so long to review it hopefully thing improve in the future. for now Good Work Ellen, if the error is fixed as soon in the picture. i will aprove. i will have a look myself this weekend if i can fix it

File: app/views/invoices/show.html.erb (hunk 1 above)
Comment (lodewiges) [reply app/views/invoices/show.html.erb:66]: could this be rewrite so it is not a "unless not"statement to improve readabilty

Comment (lodewiges) [reply app/views/invoices/show.html.erb:66]:
```suggestion
      <% if policy(Invoice).send_invoice? && !@invoice.paid? %>
```

File: app/views/index/index.html.erb [hunk 5]
```diff
@@ -3,7 +3,7 @@
 
 <div class="container py-sm-3">
   <div class="row">
-    <% if current_user&.treasurer? || current_user&.main_bartender? %>
+    user&.treasurer? || user&.renting_manager? || user&.main_bartender?
```
Comment (lodewiges) [inline app/views/index/index.html.erb:6]:
```suggestion
    <% if user&.treasurer? || user&.renting_manager? || user&.main_bartender? %>
```

**Review by lodewiges — DISMISSED** [review]

**Comment by lodewiges** [comment]
nvm my bug fix broke the tests

**Comment by lodewiges** [comment]
i will fix this next week

**Review by lodewiges — APPROVED** [review]

**Comment by lodewiges** [comment]
I implemented some changes, mostly small fixes to improve readability. and fixed an error due to an unpermitted parameter. I also changed the default text in the search bar on the product totals table to be more intuitive. Cause I was unable to determine what I could look op using the search bar.

**Pull request merged**